crates/shared/src/sources/uniswap_v2/pool_fetching.rs

```diff
@@ -54,6 +52,17 @@ pub trait PoolReading: Sized + Send + Sync {
     ) -> BoxFuture<'_, Result<Option<Pool>>>;
 }
 
+impl PoolReading for Box<dyn PoolReading> {
+    fn read_state(
+        &self,
+        pair: TokenPair,
+        batch: &mut Web3CallBatch,
+        block: BlockId,
+    ) -> BoxFuture<'_, Result<Option<Pool>>> {
+        (**self).read_state(pair, batch, block)
+    }
+}
```

Comment:
Do we really need `Arc<Box>`? This seems weird.

Reply:
There is no Arc<Box>